packages/outstatic/src/pages/add-custom-field/index.tsx

```diff
@@ -0,0 +1,557 @@
+
+    if (!selectedField && customFields[fieldName]) {
+      methods.setError('title', {
+        type: 'manual',
+        message: 'Field name is already taken.'
+      })
+      setAdding(false)
+      return
+    }
+
+    try {
+      customFields[fieldName] = {
+        ...rest,
+        title: capitalize(data.title)
+      }
+
+      const created = await capiHelper({ customFields })
+
+      if (created) {
+        setCustomFields({ ...customFields })
+      }
+    } catch (error) {
+      // TODO: Better error treatment
+      setError('add')
+      console.log({ error })
+    }
+
+    setHasChanges(false)
+    setSelectedField('')
+    setShowAddModal(false)
+    setAdding(false)
+  }
+
+  const deleteField = async (name: string) => {
+    setDeleting(true)
+
+    try {
+      let newCustomFields = { ...customFields }
+      delete newCustomFields[name]
+      const deleted = await capiHelper({
+        customFields: newCustomFields,
+        deleteField: true
+      })
+
+      if (deleted) {
+        setCustomFields(newCustomFields)
+      }
+    } catch (error) {
+      // TODO: Better error treatment
+      setError('delete')
+      console.log({ error })
+    }
+    setSelectedField('')
+    setShowDeleteModal(false)
+    setDeleting(false)
+    setHasChanges(false)
+  }
+
+  useEffect(() => {
+    const documentQueryObject = schemaQueryData?.repository?.object
+    if (documentQueryObject?.__typename === 'Blob') {
+      const schema = JSON.parse(documentQueryObject?.text || '{}')
+      setCustomFields(schema.properties)
+    }
+  }, [schemaQueryData])
+
+  useEffect(() => {
+    const subscription = methods.watch(() => setHasChanges(true))
+    return () => subscription.unsubscribe()
+  }, [methods])
+
+  // Ask for confirmation before leaving page if changes were made.
+  useNavigationLock(hasChanges)
+
+  return (
+    <FormProvider {...methods}>
+      <AdminLayout title="Add Custom Fields">
+        <div className="mb-8 flex h-12 items-center">
+          <h1 className="mr-12 text-2xl">
+            <span className="capitalize">{collection}</span> Fields
+          </h1>
+          {Object.keys(customFields).length > 0 ? (
+            <button
+              type="button"
+              onClick={() => {
+                methods.reset()
+                setShowAddModal(true)
+              }}
+            >
+              <div className="cursor-pointer rounded-lg border px-5 py-2.5 text-sm font-medium focus:outline-none focus:ring-4 border-gray-600 bg-gray-800 text-white hover:border-gray-600 hover:bg-gray-700 focus:ring-gray-700 no-underline">
+                Add Custom Field
+              </div>
+            </button>
+          ) : null}
+        </div>
+        {!loading ? (
+          <>
+            {Object.keys(customFields).length === 0 ? (
+              <div className="max-w-2xl">
+                <div className="absolute bottom-0 left-0 md:left-64 right-0 md:top-36">
+                  <svg
```

Comment:
Have we reached a point where these SVG should be in their own `src/icons` directory or equivalent?

Reply:
Probably, but for the sake of keeping this PR small we can tackle this later. This SVG background is only used in 3 files so far.